integration-tests/http/__tests__/product/admin/product-export.spec.ts

```diff
@@ -0,0 +1,398 @@
+import { medusaIntegrationTestRunner } from "medusa-test-utils"
+import {
+  adminHeaders,
+  createAdminUser,
+} from "../../../../helpers/create-admin-user"
+import { getProductFixture } from "../../../../helpers/fixtures"
+
+jest.setTimeout(50000)
+
+medusaIntegrationTestRunner({
+  testSuite: ({ dbConnection, getContainer, api }) => {
+    let baseProduct
+    let proposedProduct
+
+    let baseCollection
+    let publishedCollection
+
+    let baseType
+
+    beforeEach(async () => {
+      await createAdminUser(dbConnection, adminHeaders, getContainer())
+
+      baseCollection = (
+        await api.post(
+          "/admin/collections",
+          { title: "base-collection" },
+          adminHeaders
+        )
+      ).data.collection
+
+      publishedCollection = (
+        await api.post(
+          "/admin/collections",
+          { title: "proposed-collection" },
+          adminHeaders
+        )
+      ).data.collection
+
+      baseType = (
+        await api.post(
+          "/admin/product-types",
+          { value: "test-type" },
+          adminHeaders
+        )
+      ).data.product_type
+
+      baseProduct = (
+        await api.post(
+          "/admin/products",
+          getProductFixture({
+            title: "Base product",
+            collection_id: baseCollection.id,
+            type_id: baseType.id,
+          }),
+          adminHeaders
+        )
+      ).data.product
+
+      proposedProduct = (
+        await api.post(
+          "/admin/products",
+          getProductFixture({
+            title: "Proposed product",
+            status: "proposed",
+            tags: [{ value: "new-tag" }],
+            type_id: baseType.id,
+          }),
+          adminHeaders
+        )
+      ).data.product
+    })
+
+    describe("POST /admin/products/export", () => {
+      it("should export a csv file containing the expected products", async () => {
+        // BREAKING: The batch endpoints moved to the domain routes (admin/batch-jobs -> /admin/products/export). The payload and response changed as well.
+        const batchJobRes = await api.post(
+          "/admin/products/export",
+          {},
+          adminHeaders
+        )
+
+        const workflowId = batchJobRes.data.workflow_id
+        expect(workflowId).toBeTruthy()
+
+        // Pull to check the status until it is completed
+        while (true) {
```

Comment:
General comment from someone that is not aware of all the underlying mechanism and not only valid for the test here:
This looks a good use case for the workflow engine and the async workflow triggers the action and we subscribe to that transaction to proceed handlw when it is done.
Even to push these events to the UI would be great.

Reply:
Yeah agreed, I just copy-pasted the tests from the original batch tests, and I will be changing them as I am implementing the exports. I'll try using what you suggested instead of the while loop 👍